components/merkledb/tests/state_aggregation.rs

```diff
@@ -109,10 +109,12 @@ fn check_patch(patch: &Patch) -> TestCaseResult {
     let aggregator = system_schema.state_aggregator();
     prop_assert_eq!(state_hash, aggregator.object_hash());
 
-    let mut absent_lists: HashSet<_> = HashSet::from_iter(LIST_NAMES.iter().map(|s| s.to_string()));
-    let mut absent_maps: HashSet<_> = HashSet::from_iter(MAP_NAMES.iter().map(|s| s.to_string()));
+    let mut absent_lists: HashSet<_> =
+        HashSet::from_iter(LIST_NAMES.iter().map(|s| (*s).to_string()));
```

Comment:
Is this some pedantic clippy lint? The change looks artificial, IMO.

Reply:
Yes, `clippy` in 1.40 complains (and this is not even `pedantic` lint):

```
Checking playground v0.0.1 (/playground)
warning: calling `to_string` on `&&str`
 --> src/main.rs:4:25
  |
4 |     list.iter().map(|s| s.to_string()).collect()
  |                         ^^^^^^^^^^^^^ help: try dereferencing the receiver: `(*s).to_string()`
  |
  = note: `#[warn(clippy::inefficient_to_string)]` on by default
  = help: `&str` implements `ToString` through a slower blanket impl, but `str` has a fast specialization of `ToString`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#inefficient_to_string
```

See [playground example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=83ccef2d786d9ad6a7f209e89e1c7507).

This is a questionable topic though: on the one hand it's an almost free possibility to write more efficient code, but on the other hand it worsens the readability.

Reply:
And one more comment: you can compare the assembly generated for both versions:

- [`s.to_string()`](https://godbolt.org/z/jc87B8) - 534 lines;
- [`(*s).to_string()`](https://godbolt.org/z/5pkaG2) - 159 lines.

Reply:
Would `|&s| s.to_string()` work in this case? IMO, it would be more readable. Using `Iterator::copied()` is another option, although if `|&s|` works, it would be optimal, IMO.

Reply:
Yep, seems that clippy is happy with `|&s|` and assembly looks identical to `(*s).to_string()`.